### PR TITLE
feat(nextra-theme-docs): bump `rehype-pretty-code` version, support `showLineNumbers`

### PR DESCRIPTION
**Pull request by B2o5T**
@shuding please review my PRs, and I'll add other ones 😉

image

image

### CODE REVIEW

File: examples/docs/next.config.mjs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import nextra from 'nextra'
```
Comment (B2o5T) [inline examples/docs/next.config.mjs:1]: it's 2022, let's use esm config

File: examples/docs/package.json [hunk 2]
```diff
@@ -11,8 +11,6 @@
     "vercel-build": "yarn workspace nextra run build && yarn workspace nextra-theme-docs run build && yarn build"
   },
   "dependencies": {
-    "prism-react-renderer": "^1.1.1",
-    "prismjs": "^1.22.0",
```
Comment (B2o5T) [inline examples/docs/package.json:15]: I guess it's deps from nextra v1

File: packages/nextra-theme-docs/src/styles.css [hunk 3]
```diff
@@ -444,16 +444,30 @@ article {
...
     code {
-      line-height: 1.25rem;
-      @apply relative p-0 text-sm text-current bg-transparent dark:bg-transparent rounded-none border-none inline-block min-w-full;
+      @apply p-0 text-sm text-current bg-transparent dark:bg-transparent rounded-none border-none grid min-w-full leading-5;
+      .line {
+        @apply px-4 border-l-2 border-transparent;
+      }
       .line.highlighted {
-        @apply before:block before:absolute before:h-5 before:bg-primary-600 before:bg-opacity-10 before:-inset-x-4 before:pointer-events-none before:select-none;
+        @apply bg-primary-600/10 border-primary-600/60;
```
Comment (B2o5T) [inline packages/nextra-theme-docs/src/styles.css:470]: improved these styles to omit `position: absolute`

File: packages/nextra/src/compile.ts [hunk 4]
```diff
@@ -21,41 +20,30 @@ const createCompiler = (mdxOptions: ProcessorOptions) => {
...
-  // },
-  onVisitHighlightedLine(node: any) {
-    // Style a highlighted line node.
-    if (!node.properties.className) {
-      node.properties.className = []
+  onVisitLine(node) {
+    // Prevent lines from collapsing in `display: grid` mode, and
+    // allow empty lines to be copy/pasted
+    if (node.children.length === 0) {
+      node.children = [{ type: 'text', value: ' ' }]
```
Comment (B2o5T) [inline packages/nextra/src/compile.ts:27]: according to docs 
image

File: examples/docs/src/pages/nextra/about.md [hunk 5]
```diff
@@ -32,10 +32,10 @@ import Callout from 'nextra-theme-docs/callout'
   MDX (the library), at its core, transforms MDX (the syntax) to JSX. It receives an MDX string and outputs a JSX string. It does this by parsing the MDX document to a syntax tree and then generates a JSX document from that tree. 
 </Callout>
 
-Code highlighting with highlighed lines using
+Code highlighting with highlighted lines using
```
Comment (B2o5T) [inline examples/docs/src/pages/nextra/about.md:35]: typo

**Comment by numToStr** [comment]
Is this change only related to `docs` theme? IMO this could also be useful for `blog` theme.

**Pull request closed without merging**